R/fda-table_34.R

```diff
@@ -0,0 +1,74 @@
+#' FDA Table 34: Patients With Serious Adverse Events1 by System Organ Class, FDA Medical Query (Narrow) and Preferred Term,
+#' Safety Population, Pooled Analysis (or Trial X)
+#'
+#' @details
+#' * `adae` must contain the variables `SAFFL`, `USUBJID`, `AEBODSYS`, and the variables specified by `arm_var`,
+#'   `fmqsc_var`, and `fmqnam_var`.
+#' * If specified, `alt_counts_df` must contain variables `SAFFL` and `USUBJID`.
+#' * Flag variables (i.e. `XXXFL`) are expected to have two levels: `"Y"` (true) and `"N"` (false). Missing values in
+#'   flag variables are treated as `"N"`.
+#' * Columns are split by arm. Overall population column is excluded by default (see `lbl_overall` argument).
+#' * Numbers in table represent the absolute numbers of patients and fraction of `N`.
+#' * All-zero rows are not removed by default (see `prune_0` argument).
+#'
+#' @inheritParams argument_convention
+#'
+#' @examples
+#' adae <- scda::synthetic_cdisc_dataset("rcd_2022_10_13", "adae")
+#' adsl <- scda::synthetic_cdisc_dataset("rcd_2022_10_13", "adsl")
+#'
+#' adae <- dplyr::rename(adae, FMQ01SC = SMQ01SC, FMQ01NAM = SMQ01NAM)
+#' levels(adae$FMQ01SC) <- c("BROAD", "NARROW")
+#' adae$FMQ01SC[is.na(adae$FMQ01SC)] <- "NARROW"
+#'
+#' tbl <- make_table_34(adae, adsl)
+#' tbl
+#'
+#' @export
+
+make_table_34 <- function(adae,
+                          alt_counts_df = NULL,
+                          show_colcounts = TRUE,
+                          arm_var = "ARM",
+                          fmqsc_var = "FMQO1SC",
+                          fmqnam_var = "FMQ01NAM",
+                          pref_var = "AETERM",
+                          lbl_pref_var = formatters::var_labels(adae, fill = TRUE)[pref_var],
+                          lbl_overall = NULL,
+                          prune_0 = FALSE,
+                          na_level = "<Missing>",
+                          annotations = NULL) {
+  checkmate::assert_subset(c("SAFFL", "USUBJID", "AEBODSYS", arm_var, fmqnam_var, fmqnam_var, pref_var), names(adae))
+  assert_flag_variables(adae, "SAFFL")
+
+  adae <- adae %>%
+    filter(SAFFL == "Y") %>%
+    filter(AESER == "Y") %>%
+    df_explicit_na(na_level = na_level)
+
+  adae <- adae[adae[[fmqsc_var]] == "NARROW", ]
+  adae[[fmqnam_var]] <- with_label(adae[[fmqnam_var]], "FMQ (Narrow)")
```

Comment:
Implement the `fmq_scope` variable (see [table 10](https://github.com/pharmaverse/falcon/blob/main/R/fda-table_10.R) which also has a `fmq_scope` parameter)
```suggestion
  checkmate::assert_subset(toupper(fmq_scope), c("NARROW", "BROAD"))

    adae <- adae %>%
      filter(SAFFL == "Y", AESER == "Y", adae[[fmqsc_var]] == fmq_scope) %>%
      df_explicit_na(na_level = na_level)

    adae[[fmqnam_var]] <- with_label(adae[[fmqnam_var]], paste0("FMQ (", tools::toTitleCase(tolower(fmq_scope)), ")"))

```